### DISCUSSION TITLE
Minify inline scripts

### DISCUSSION THREAD

**Discussion by dkobierski** [withastro/roadmap#799] [Proposal]
Currently when you want to have an inline script you need to add `is:inline` directive to the script. This makes this script to be untouched by the astro meaning it won't be bundled so you can't use typescript or imports there, but this script is also not minified/compressed which affects performance and also makes the rendered html look weird when compressHtml is enabled (everything but scripts is compressed). It would be nice to have an option to minify the inline js as well.

**Comment by OliverSpeir** [comment]
minify and allow typescript is what I'd like to see!

**Comment by azat-io** [comment]
This would be a useful feature. 👍

**Comment by delucis** [comment]
Here’s another related proposal: https://github.com/withastro/roadmap/discussions/995
